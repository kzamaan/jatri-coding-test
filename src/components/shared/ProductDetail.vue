<script setup>
    defineProps({
        productDetail: {
            type: Object,
            default: () => ({}),
        },
        isLoading: {
            type: Boolean,
            default: false,
        },
    });
</script>

<template>
    <tr>
        <td colspan="5">
            <div v-if="isLoading">Loading</div>
            <div>Product Detail</div>
        </td>
    </tr>
</template>
