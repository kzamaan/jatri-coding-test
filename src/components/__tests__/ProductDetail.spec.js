import { describe, expect, it } from 'vitest';

import { mount } from '@vue/test-utils';
import ProductDetail from '../shared/ProductDetail.vue';

describe('ProductDetail', () => {
    it('renders product detail properly', () => {
        const wrapper = mount(ProductDetail, {
            props: {
                productDetail: {
                    id: 1,
                    title: 'iPhone 9',
                    description: 'An apple mobile which is nothing like apple',
                    price: 549,
                    discountPercentage: 12.96,
                    rating: 4.69,
                    stock: 94,
                    brand: 'Apple',
                },
                isLoading: false,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
