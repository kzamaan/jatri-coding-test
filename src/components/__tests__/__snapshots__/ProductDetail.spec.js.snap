// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductDetail > renders product detail properly 1`] = `
"<tr data-v-8c663632=\\"\\">
  <td data-v-8c663632=\\"\\" colspan=\\"5\\">
    <!-- added slide down transition -->
    <transition-stub data-v-8c663632=\\"\\" name=\\"slide-fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
      <div data-v-8c663632=\\"\\" class=\\"card mb-2\\">
        <div data-v-8c663632=\\"\\" class=\\"card-header\\">
          <h3 data-v-8c663632=\\"\\" class=\\"text-xl font-semibold text-gray-900 dark:text-white text-center\\">Product Detail</h3>
        </div>
        <div data-v-8c663632=\\"\\" class=\\"card-body\\">
          <table data-v-8c663632=\\"\\" class=\\"w-full\\">
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Name:</td>
              <td data-v-8c663632=\\"\\">iPhone 9</td>
            </tr>
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Price:</td>
              <td data-v-8c663632=\\"\\">549</td>
            </tr>
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Rating:</td>
              <td data-v-8c663632=\\"\\">4.69</td>
            </tr>
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Description:</td>
              <td data-v-8c663632=\\"\\">An apple mobile which is nothing like apple</td>
            </tr>
          </table>
        </div>
      </div>
    </transition-stub>
  </td>
</tr>"
`;

exports[`ProductDetail > renders product properly 1`] = `
"<tr data-v-8c663632=\\"\\">
  <td data-v-8c663632=\\"\\" colspan=\\"5\\">
    <!-- added slide down transition -->
    <transition-stub data-v-8c663632=\\"\\" name=\\"slide-fade\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
      <div data-v-8c663632=\\"\\" class=\\"card mb-2\\">
        <div data-v-8c663632=\\"\\" class=\\"card-header\\">
          <h3 data-v-8c663632=\\"\\" class=\\"text-xl font-semibold text-gray-900 dark:text-white text-center\\">Product Detail</h3>
        </div>
        <div data-v-8c663632=\\"\\" class=\\"card-body\\">
          <table data-v-8c663632=\\"\\" class=\\"w-full\\">
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Name:</td>
              <td data-v-8c663632=\\"\\">iPhone 9</td>
            </tr>
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Price:</td>
              <td data-v-8c663632=\\"\\">549</td>
            </tr>
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Rating:</td>
              <td data-v-8c663632=\\"\\">4.69</td>
            </tr>
            <tr data-v-8c663632=\\"\\">
              <td data-v-8c663632=\\"\\" class=\\"font-bold\\">Product Description:</td>
              <td data-v-8c663632=\\"\\">An apple mobile which is nothing like apple</td>
            </tr>
          </table>
        </div>
      </div>
    </transition-stub>
  </td>
</tr>"
`;
